tests/SilverStripeRemoteRepositoryTest.php

```diff
@@ -3,75 +3,161 @@
+        file_put_contents($filePath, json_encode([
+            "silverstripe/admin" => [
+                "repository" => "https://github.com/silverstripe/silverstripe-admin.git",
+                "versionmap" => [
+                    "5" => "2",
+                    "4" => "1",
+                    "3" => null,
+                ],
+            ],
+            "silverstripe/asset-admin" => [
+                "repository" => "https://github.com/silverstripe/silverstripe-asset-admin.git",
+                "versionmap" => [
+                    "5" => "2",
+                    "4" => "1",
+                    "3" => null,
+                ],
+            ],
+            "silverstripe/assets" => [
+                "repository" => "https://github.com/silverstripe/silverstripe-assets.git",
+                "versionmap" => [
+                    "5" => "2",
+                    "4" => "1",
+                    "3" => null,
+                ],
+            ],
+            "silverstripe/cms" => [
+                "repository" => "https://github.com/silverstripe/silverstripe-cms.git",
+                "versionmap" => [
+                    "5" => "5",
+                    "4" => "4",
+                    "3" => "3",
+                ],
+            ],
+            "silverstripe/config" => [
+                "repository" => "https://github.com/silverstripe/silverstripe-config.git",
+                "versionmap" => [
+                    "5" => "2",
+                    "4" => "1",
+                    "3" => "1",
+                ],
+            ],
+            "silverstripe/framework" => [
+                "repository" => "https://github.com/silverstripe/silverstripe-framework.git",
+                "versionmap" => [
+                    "5" => "5",
+                    "4" => "4",
+                    "3" => "3",
+                ],
+            ],
+            "silverstripe/graphql" => [
+                "repository" => "https://github.com/silverstripe/silverstripe-graphql.git",
+                "versionmap" => [
+                    "5" => "5",
+                    "4" => "4",
+                    "3" => null,
+                ],
+            ],
+        ], JSON_PRETTY_PRINT | JSON_UNESCAPED_SLASHES | JSON_UNESCAPED_UNICODE));
+    }
 
-        $this->assertSame(
-            'https://github.com/silverstripe/'
-            . 'silverstripe-graphql/blob/5/src/Extensions/DevBuildExtension.php#L0',
-            $url
-        );
+    public static function tearDownAfterClass(): void
+    {
+        $config = Config::getConfig();
+        $filePath = Config::configPath($config['paths']['versionmap']);
+        unlink($filePath);
+        if (self::$fileContent !== null) {
+            file_put_contents($filePath, self::$fileContent);
+        }
+    }
 
-        $url = $remoteRepo->getFileUrl(
-            '3',
-            'silverstripe/graphql/src/Extensions/IntrospectionProvider.php',
-            0
-        );
+    public function provideGetRepoUrl()
```

Comment:
These are all the same tests that were happening before (except the graphql 3 one which was silly - that's not a valid mapping) plus one for CMS 5 and 3 just so we have one for each of the three mapped versions.